library/scrollphathd/api/api.py

```diff
@@ -0,0 +1,73 @@
+from sys import exit
+import scrollphathd
+from scrollphathd.fonts import font3x5
+from queue import Queue
+from threading import Thread
+from .action import Action
+from http import HTTPStatus
+
+try:
+    from flask import Blueprint, render_template, abort, request
+except ImportError:
+    exit("flask must be installed in order to use the api. Install with pip install flask")
+
+scrollphathd_blueprint = Blueprint('scrollhat', __name__)
+api_queue = Queue()
+
+@scrollphathd_blueprint.route('/scroll', methods=["POST"])
+def scroll():
+    data = request.get_json()
+    try:
+        api_queue.put(Action("scroll", (data["x"], data["y"])))
+    except KeyError:
+        return {"error": "keys x and y not posted."}, HTTPStatus.UNPROCESSABLE_ENTITY.value
+    else:
+        return HTTPStatus.OK.value
+
+
+@scrollphathd_blueprint.route('/show', methods=["POST"])
+def show(text):
+    data = request.get_json()
+    try:
+        api_queue.put(Action("write", data["text"]))
+    except KeyError:
+        return {"error": "key 'text' not set"}, HTTPStatus.UNPROCESSABLE_ENTITY.value
```

Comment:
Not a change, but I just wanted to point out that `HTTPStatus.STATUS.vaue` is another new thing I've learned from this PR. Thanks!

Reply:
It makes code a bit waffly sometimes, but I never remember all the status codes so means I don’t have to go looking them up :)